git-ssh/git-clone-ssh.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: git-clone-ssh
+spec:
+  workspaces:
+    - name: output
+      description: The git repo will be cloned onto the volume backing this workspace
+  params:
+    - name: url
+      type: string
+      description: git url to clone
+  steps:
+    - name: git-clone
+      image: bitnami/git:2.26.2
+      command: ['git', '-c', 'core.sshCommand=ssh -i /etc/ssh/id_rsa', 'clone', '$(params.url)', '$(workspaces.output.path)']
+      volumeMounts:
+        - mountPath: /etc/ssh
+          name: github-ssh-auth
+  volumes:
+    - name: github-ssh-auth
+      projected:
+        defaultMode: 0400
+        sources:
+          - secret:
+              name: github-known-hosts
+          - secret:
+              name: github-private-key
```

Comment:
both secret names should be _params_ in a proper implementation.

Reply:
Is this a limitation of Tekton at the moment - params can't be used as variables in volumes?

Reply:
Yes, it is the Secret Name within a projected volume, were we don't have implemented _variable substitution_. But as the commit is implemented now, it is working. But for proper support, we should add _variable substitution_ here, so that _end-users_ can name the secrets how they want.